test/service.spec.js

```diff
@@ -1,16 +1,48 @@
 // Generated by CoffeeScript 1.12.3
 (function() {
-  var AWS, MockService, helpers, metadata,
+  var AWS, MockService, helpers, metadata, MockServiceWithErrorCodeMapping
     hasProp = {}.hasOwnProperty;
 
   helpers = require('./helpers');
 
   AWS = helpers.AWS;
 
   MockService = helpers.MockService;
+  MockServiceWithErrorCodeMapping = helpers.MockServiceWithErrorCodeMapping;
 
   metadata = require('../apis/metadata.json');
 
+  describe('AWSQuery compatible error codes', function () {
+    var service;
+    var mapping = { ServiceError: { code: 'Custom.ErrorCode' } };
+    beforeEach(function() {
+      var CustomService = MockServiceWithErrorCodeMapping(mapping);
+      service = new CustomService({maxRetries: 0});
```

Comment:
could we inline `CustomService` here instead?

Reply:
I think it's clearer this way.